homeassistant/components/flunearyou/diagnostics.py

```diff
@@ -0,0 +1,40 @@
+"""Diagnostics support for Flu Near You."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.diagnostics import async_redact_data
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_LATITUDE, CONF_LONGITUDE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import CATEGORY_CDC_REPORT, CATEGORY_USER_REPORT, DOMAIN
+
+TO_REDACT = {
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+}
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+    coordinators: dict[str, DataUpdateCoordinator] = hass.data[DOMAIN][entry.entry_id]
+
+    return {
+        "entry": {
+            "title": entry.title,
+            "data": async_redact_data(entry.data, TO_REDACT),
+        },
```

Comment:
```suggestion
```

The title and the data both exist of only lat_long, I suggest not adding the entry details at all for this integration

Reply:
in this case I would suggest to still add the complete entry, but redact the title and both lat and long values.
So you have still infos about system config and source of the config entry in the diag output, which might be helpful.

Reply:
The source can only be "user"

Reply:
ok, but you have still left both the system options, which still could be very helpful
![image](https://user-images.githubusercontent.com/35783820/150805930-2c79d8ed-06a2-4c56-aaff-a493900ca6b8.png)

Reply:
Those will never need redaction, and should be added to the base response in that case

Reply:
> should be added to the base response in that case

great idea 💡 so lets extend the diagnostics platform to provide this (_as already done for hass system info, custom components and manifest_)?

Reply:
Going to resolve this conversation since this particular PR's feedback has been incorporated.